src/core/annotation.js

```diff
@@ -1066,14 +1066,21 @@ class WidgetAnnotation extends Annotation {
     }
 
     let loopDict = dict;
+    const visited = new Set([this.data.id]);
```

Comment:
First of all, please use a `RefSet` instead. (Ideally the Dictionary instances would actually use a `Ref` directly, rather than their string-format, so let's avoid future surprises here.)

Secondly, `this.data.id` isn't guaranteed to contain the string-format of a `Ref`; please see https://github.com/mozilla/pdf.js/blob/a392082e302366a25a17d1ba35551a2acd9e5195/src/core/annotation.js#L82, hence you should use `dict.objId` instead.

Reply:
I don't see any advantage to use a `RefSet` which seems to be useful to store a `Ref`. Here `objId` is a string or null (correct ?).

Reply:
Sorry about the delay here, I got sidetracked with other things lately!

---

The fact that `Dictionary`-instance uses a string for the `objId`, rather than simply storing the `Ref` directly, is the result of some *really* old implementation decisions. Hence we might, at some point, actually want to fix that and then it's simpler if all code actually handles (what should be) `Ref`s consistently.